### PR TITLE
Modified voters/deciders to follow scikit-learn API's and edited test files to reflect these changes

### LINKED ISSUE #226: Modify Classification {Voters, Deciders} to follow scikit-learn {ClassifierMixin, BaseEstimator} API's
Guide on how to comply a class with a scikit-learn API: https://sklearn-template.readthedocs.io/en/latest/user_guide.html

ClassifierMixin docs: https://scikit-learn.org/stable/modules/generated/sklearn.base.ClassifierMixin.html#sklearn.base.ClassifierMixin

BaseEstimator docs: https://scikit-learn.org/stable/modules/generated/sklearn.base.BaseEstimator.html#sklearn.base.BaseEstimator

**BaseEstimator**
The following classes should inherit from BaseEstimator explicitly in their class names:
1. BaseVoter in base.py
2. BaseDecider in base.py

For example, you should be changing the class name of 'BaseVoter' to 'BaseVoter(BaseEstimator)'. 

**ClassifierMixin**
The following classes should inherit from ClassifierMixin explicitly in their class names:
1. TreeClassificationVoter in voter.py
2. KNNClassificationVoter in voter.py
3. BaseClassificationDecider in decider.py

For example, you should be changing the class name of 'TreeClassificationVoter(BaseVoter)' to 'TreeClassificationVoter(BaseVoter, BaseEstimator, ClassifierMixin)' 

The two important functions for ClassifierMixin are fit, predict, and predict_proba. 
1. fit: All of these classes already have fit methods, so you don't need to worry about that. 
2. predict_proba: BaseClassificationDecider (and its subclasses) already have predict and predict_proba, so you should just need to add 'ClassifierMixin' as a superclass in the BaseClassificationDecider class name. The voter predict_proba function is called "vote". So you'll have to change the name of the method from "vote" to "predict_proba".
3. predict: The voter currently doesn't have a "predict" function. So you'll have to make one. I would simply call the 'predict_proba' method (what is currently called 'vote') and return the argmax of each index. This is as simple as a one-line 'return np.argmax(self.predict_proba(X), axis = 1)'

**IMPORTANT NOTES**
Please please please please remember to reflect these changes in the docstrings of every class that you change. Any time you change a function name, please reflect that in the docstring for the class to which that function belonged. Any time you subclass from something, please reflect that in the docstring for that class.

You will also have to modify the ProgressiveLearner class in progressive_learner.py to reflect function name changes, namely from 'voter.vote' to 'voter.predict_proba'.

### PR DESCRIPTION
**Pull request by emilyachang**
Closes #226 with the modifications specified in the issue description + additional changes to tests/test_voter.py to account for renaming of voter.vote to voter.predict_proba

### CODE REVIEW

File: proglearn/deciders.py [hunk 1]
```diff
@@ -17,8 +17,10 @@
 
 from sklearn.utils.multiclass import type_of_target
 
+from sklearn.base import BaseEstimator, ClassifierMixin
 
-class SimpleArgmaxAverage(BaseClassificationDecider):
+
+class SimpleArgmaxAverage(BaseClassificationDecider, BaseEstimator, ClassifierMixin):
```
Comment (levinwil) [inline proglearn/deciders.py:23]: You don't need to interit from BaseEstimator or ClassifierMixin. By nature of inserting from BaseClassificationDecider, it inherits from BaseDeicder, which inerits from BaseEstimator. Also, if you make BaseClassificationDecider from ClassifierMixin, then SimpleArgmaxAverage will inherit from ClassifierMixin, since it inherits from BaseClassificationDecider.

File: proglearn/voters.py [hunk 2]
```diff
@@ -14,12 +14,15 @@
 
 from sklearn.utils.multiclass import check_classification_targets
 
+from sklearn.base import BaseEstimator, ClassifierMixin
+
 from .base import BaseVoter
 
 
-class TreeClassificationVoter(BaseVoter):
+class TreeClassificationVoter(BaseVoter, BaseEstimator, ClassifierMixin):
```
Comment (levinwil) [inline proglearn/voters.py:22]: You don't need to inherit from BaseEstimator. By nature of inheriting from BaseVoter, it inherits from BaseEstimator.

File: proglearn/voters.py [hunk 3]
```diff
@@ -137,10 +154,11 @@ def _finite_sample_correction(posteriors, num_points_in_partition, num_classes):
         return posteriors
 
 
-class KNNClassificationVoter(BaseVoter):
+class KNNClassificationVoter(BaseVoter, BaseEstimator, ClassifierMixin):
```
Comment (levinwil) [inline proglearn/voters.py:157]: You don't need to inherit from BaseEstimator. By nature of inheriting from BaseVoter, it inherits from BaseEstimator.

**Review by levinwil — COMMENTED** [review]
I added some comments about inheritance. One thing I would change as well: in Base.py, you'll see there is a BaseDecider and BaseClassificationDecider, where the classification one adds predict_proba. I would follow the same steps for the voter, making a base classification voter that inherits from base voter. I would inherit base voter from sklearn.BaseEstimator and BaseClassificationVoter from BaseVoter and from sklearn.ClassifierMixIn. Then all of the voters that you changed should inherit from BaseClassificationVoter (and just BaseClassificationVoter, not any of the sklearn interfaces directly).

**Comment by emilyachang** [comment]
@levinwil Made the suggested changes -- could you take a look?

**Comment by levinwil** [comment]
@emilyachang Looks great!

**Review by levinwil — APPROVED** [review]
